.gitlab/ci/on-push.yml

```diff
@@ -76,183 +75,144 @@ validate-content-conf:
 
-create-instances:
+.generic-prepare-testing-bucket:
   tags:
     - gke
   extends:
     - .default-job-settings
   rules:
-    - if: '$CI_PIPELINE_SOURCE =~ /^(push|contrib)$/'
-    - if: '$NIGHTLY'
-    - if: '$INSTANCE_TESTS'
+    - if: "$CI_PIPELINE_SOURCE =~ /^(push|contrib)$/"
+    - if: "$NIGHTLY"
   cache:
     policy: pull-push
   variables:
-    ARTIFACTS_FOLDER: "${ARTIFACTS_FOLDER_XSOAR}"
     KUBERNETES_CPU_REQUEST: 2000m
   needs: []
-  stage: create-instances
+  stage: prepare-testing-bucket
   script:
     - !reference [.download-demisto-conf]
-    - !reference [.create-id-set-xsoar]
     - !reference [.create-release-notes-and-common-docs]
 
+    - section_start "Create or update content graph" --collapsed
+
+    - |
+      echo "Trying to pull all CSV import files from bucket"
+      gcloud auth activate-service-account --key-file="$GCS_MARKET_KEY" > auth.out 2>&1
+      gsutil -m cp -r "gs://$GCS_MARKET_BUCKET_DEV/content_graph/$MARKETPLACE_VERSION/*" /var/lib/neo4j/import || echo "failed to download graph imports."
+      echo "Successfully pulled all CSV import files from bucket"
+    - |
+      if [ -z "$(ls -A /var/lib/neo4j/import)" ]; then
+        echo "No CSV files to import, creating the content graph"
+        demisto-sdk create-content-graph
+        echo "Successfully created content graph"
+      else
+        echo "Getting the list of packs to update in content graph"
+        PACKS_TO_UPDATE=( $(git diff $LAST_UPLOAD_COMMIT --name-only -- Packs/* | awk -F/ '{print $2}' | sort | uniq) )
+        echo "Updating graph for packs: $PACKS_TO_UPDATE"
+
+        echo "Updating the content graph"
+        demisto-sdk update-content-graph -p $PACKS_TO_UPDATE
+        echo "Successfully updated content graph"
+      fi
+
+    - |
+      echo "Moving exported CSV files to artifacts"
+      mkdir $ARTIFACTS_FOLDER/content_graph
+      cp -R /var/lib/neo4j/import $ARTIFACTS_FOLDER/content_graph
+
+    - section_end "Update content graph"
+
     - section_start "Create Content Artifacts and Update Conf" --collapsed
-    - demisto-sdk create-content-artifacts -a $ARTIFACTS_FOLDER --cpus 8 --content_version $CONTENT_VERSION --marketplace "xsoar" -fbi -idp ./Tests/id_set.json >> $ARTIFACTS_FOLDER/logs/create_content_artifacts.log
+    - |
+      if [[ $MARKETPLACE_VERSION == "xsoar" ]];
+      then
+        echo "Starting to create artifacts with zip for XSOAR."
+        python Tests/scripts/create_artifacts_graph/create_artifacts.py --marketplace "xsoar" --artifacts-output $ARTIFACTS_FOLDER/content_packs --dependencies-output $ARTIFACTS_FOLDER/packs_dependencies.json
+      else
+        echo "Starting to create artifacts without zip."
+        python Tests/scripts/create_artifacts_graph/create_artifacts.py --marketplace "$MARKETPLACE_VERSION" --artifacts-output $ARTIFACTS_FOLDER/content_packs --dependencies-output $ARTIFACTS_FOLDER/packs_dependencies.json --no-zip
+      fi
```

Comment:
why do we need to distinguish?

Reply:
with non-xsoar we use the `--no-zip` argument